File: src/main/scala/cc/sukazyo/hytrans_intellij/language/HytTranslationsFileType.scala
package cc.sukazyo.hytrans_intellij.language

import cc.sukazyo.hytrans_intellij.Resources
import com.intellij.openapi.fileTypes.LanguageFileType
import org.jetbrains.annotations.NotNull

import javax.swing.Icon

class HytTranslationsFileType private extends LanguageFileType (HytTranslationsLanguage.INSTANCE) {
	
	@NotNull override def getName: String = "HyTrans Translation File"
	@NotNull override def getDescription: String = "HyTrans translation file contains key-value pairs translation definition."
	@NotNull override def getDefaultExtension: String = "hytrans"
	@NotNull override def getIcon: Icon = Resources.ICON_HYTRANS_FILE
	
}

object HytTranslationsFileType {
	val INSTANCE = new HytTranslationsFileType
}

File: src/main/scala/cc/sukazyo/hytrans_intellij/language/HytTranslationsFileType.java
package cc.sukazyo.hytrans_intellij.language;

import cc.sukazyo.hytrans_intellij.Resources$;
import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.util.NlsContexts;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class HytTranslationsFileType extends LanguageFileType {
	
	private HytTranslationsFileType() { super(HytTranslationsLanguage.INSTANCE); }
	public static final HytTranslationsFileType INSTANCE = new HytTranslationsFileType();
	
	@Override public @NonNls @NotNull String getName () { return "HyTrans Translation File"; }
	@Override public @NlsContexts.Label @NotNull String getDescription () { return "HyTrans translation file contains key-value pairs translation definition."; }
	@Override public @NonNls @NotNull String getDefaultExtension () { return "hytrans"; }
	@Override public @NonNls Icon getIcon () { return Resources$.MODULE$.ICON_HYTRANS_FILE(); }
	
}
